tiledb/cloud/notebook.py

```diff
@@ -35,3 +50,290 @@ def rename_notebook(
+    :param ctx: cloud context for the operation.
+    :return: tuple of tiledb_uri and array_name
+    """
+
+    if namespace in RESERVED_NAMESPACES:
+        raise ValueError(
+            f"{namespace!r} is not a valid folder to create notebooks. "
+            "Please select a proper namespace (username or organization name).",
+        )
+
+    # The array will be be 1-dimensional with domain of 0 to max uint64. We
+    # use a tile extent of 1024 bytes.
+    dom = tiledb.Domain(
+        tiledb.Dim(
+            name="position",
+            domain=(0, numpy.iinfo(numpy.uint64).max - 1025),
+            tile=1024,
+            dtype=numpy.uint64,
+            ctx=ctx,
+        ),
+        ctx=ctx,
+    )
+
+    tries = 1 + retries  # 1st + rest
+    while tries > 0:
+        try:
+            tiledb_uri, array_name = _create_notebook_array_retry_helper(
+                storage_path,
+                array_name,
+                namespace,
+                dom,
+                ctx,
+            )
+            return (tiledb_uri, array_name)
+        except tiledb.TileDBError as e:
+            if "Error while listing with prefix" in str(e):
+                # It is possible to land here if user sets wrong default S3
+                # credentials with respect to default S3 path.
+                raise ValueError(
+                    f"Error creating file: {e}. Are your credentials valid?"
+                ) from e
+            if "Cannot create array" in str(e) and "already exists" in str(e):
+                raise ValueError(
+                    f"Error creating file: {array_name} already exists in namespace {namespace}."
+                )
+        # Retry other exceptions
+        tries -= 1
+    raise ValueError(f"Out of retries uploading {array_name} to namespace {namespace}.")
+
+
+def _create_notebook_array_retry_helper(
+    storage_path: str,
+    array_name: str,
+    namespace: str,
+    dom: tiledb.Domain,
+    ctx: tiledb.Ctx,
+) -> Tuple[bool, str, str]:
+    """
+    See _create_notebook_array -- exists only for retry logic.
+    :return: tuple of succeeded, tiledb_uri, and array_name
+    """
+
+    schema = tiledb.ArraySchema(
+        domain=dom,
+        sparse=False,
+        attrs=[
+            tiledb.Attr(
+                name="contents",
+                dtype=numpy.uint8,
+                filters=tiledb.FilterList([tiledb.ZstdFilter()]),
+            )
+        ],
+        ctx=ctx,
+    )
+
+    # Goal: tiledb://my_username/s3://my_bucket/my_array
+    # https://docs.tiledb.com/cloud/how-to/arrays/create-arrays
+    tiledb_uri_s3 = "tiledb://" + posixpath.join(namespace, storage_path, array_name)
+
+    # Create the (empty) array on disk.
+    tiledb.DenseArray.create(tiledb_uri_s3, schema)
+    tiledb_uri = "tiledb://" + posixpath.join(namespace, array_name)
+    time.sleep(0.25)
+
+    file_properties = {}
+
+    tiledb.cloud.array.update_info(uri=tiledb_uri, array_name=array_name)
```

Comment:
Since you've `from tiledb.cloud import array`, you should use `array.update_info` here.